src/gurt/telemetry.c

```diff
@@ -763,65 +778,69 @@ d_tm_compute_standard_dev(double sum_of_squares, uint64_t sample_size,
 }
 
 /**
- * Compute gauge statistics: sample size, min, max, sum and sum of squares.
+ * Compute statistics: sample size, min, max, sum and sum of squares.
  * Standard deviation calculation is deferred until the metric is read.
  *
- * \param[in]	node		Pointer to a gauge node
+ * \param[in]	node		Pointer to a node with stats
  */
 void
-d_tm_compute_gauge_stats(struct d_tm_node_t *node)
+d_tm_compute_stats(struct d_tm_node_t *node, uint64_t value)
 {
 	struct d_tm_stats_t	*dtm_stats;
-	uint64_t		value = 0;
 
 	dtm_stats = node->dtn_metric->dtm_stats;
 
 	if (dtm_stats == NULL)
 		return;
 
-	value = node->dtn_metric->dtm_data.value;
 	dtm_stats->sample_size++;
-	dtm_stats->dtm_sum.sum_int += value;
+	dtm_stats->dtm_sum += value;
 	dtm_stats->sum_of_squares += value * value;
 
-	if (value > dtm_stats->dtm_max.max_int)
-		dtm_stats->dtm_max.max_int = value;
+	if (value > dtm_stats->dtm_max)
+		dtm_stats->dtm_max = value;
 
-	if (value < dtm_stats->dtm_min.min_int)
-		dtm_stats->dtm_min.min_int = value;
+	if (value < dtm_stats->dtm_min)
+		dtm_stats->dtm_min = value;
 }
 
 /**
- * Compute duration statistics: sample size, min, max, sum and sum of squares.
- * Standard deviation calculation is deferred until the metric is read.
+ * Computes the histogram for this metric by finding the bucket that corresponds
+ * to the \a value given, and increments the counter for that bucket.
  *
- * \param[in]	node		Pointer to a duration node
+ * \param[in]	node		Pointer to a duration or gauge node
+ * \param[in]	value		The value that is sorted into a bucket
  */
-void
-d_tm_compute_duration_stats(struct d_tm_node_t *node)
+int
+d_tm_compute_histogram(struct d_tm_node_t *node, uint64_t value)
 {
-	struct d_tm_stats_t	*dtm_stats;
-	double			value = 0;
-
-	dtm_stats = node->dtn_metric->dtm_stats;
+	struct d_tm_histogram_t	*dtm_histogram;
+	struct d_tm_node_t	*bucket;
+	int			rc = D_TM_SUCCESS;
+	int			i;
 
-	if (dtm_stats == NULL)
-		return;
+	if (!node)
+		return rc;
 
-	value = node->dtn_metric->dtm_data.tms[0].tv_sec +
-		(node->dtn_metric->dtm_data.tms[0].tv_nsec / 1E9);
+	if (!node->dtn_metric)
+		return rc;
 
-	dtm_stats->sample_size++;
-	dtm_stats->dtm_sum.sum_float += value;
-	dtm_stats->sum_of_squares += value * value;
+	if (!node->dtn_metric->dtm_histogram)
+		return rc;
```

Comment:
Hm, do we want to ignore these? Or return -DER_INVAL instead?

Reply:
I'll go with -DER_INVAL for now, and make the changes to return void where we can in another PR.